File: backend/src/main/java/com/apocalypse/thefall/service/EventService.java
package com.apocalypse.thefall.service;

import com.apocalypse.thefall.event.CharacterMovementEvent;
import lombok.RequiredArgsConstructor;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class EventService {

    private final KafkaTemplate<String, CharacterMovementEvent> kafkaTemplate;

    public void publishMovementEvent(CharacterMovementEvent event) {
        kafkaTemplate.send("player-movement", event);
    }
}

File: backend/src/main/java/com/apocalypse/thefall/service/EventService.kt
package com.apocalypse.thefall.service

import com.apocalypse.thefall.event.CharacterMovementEvent
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
open class EventService(
    private val kafkaTemplate: KafkaTemplate<String, CharacterMovementEvent>
) {

    fun publishMovementEvent(event: CharacterMovementEvent) {
        kafkaTemplate.send("player-movement", event)
    }
}
